### PR TITLE
feature: add async Block-powered "Examples" page

### PR DESCRIPTION
**Pull request by bitpshr**
## Description

This pull request adds an "Examples" page that's powered by an asynchronous `Block` that pulls data from the [dojo/examples](https://github.com/dojo/examples) README. The design could use some iteration, but the necessary pieces to dynamically build example content are in-place. Existing styles and widgets were leveraged as much as possible.

### Code

This PR touches:

- [x] Content
- [x] Frontend
- [x] Infrastructure
- [x] Integration Tests

### Tests

- [x] Tests are included?

### Screenshots

preview

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# http://editorconfig.org
```
Comment (bitpshr) [inline .editorconfig]: Added for my own sanity; I work in several projects with different defaults, and I don't lint or prettify on save.

File: src/pages/Examples.spec.tsx [hunk 2]
```diff
@@ -1,16 +1,94 @@
...
+import FontAwesomeIcon from '../widgets/icon/FontAwesomeIcon';
+import Grid from '../widgets/grid/Grid';
+import Landing from '../widgets/Landing';
+import LinkedCard from '../widgets/card/LinkedCard';
+import Subsection from '../widgets/section/Subsection';
 
 import Examples from './Examples';
 import * as css from './Examples.m.css';
 
+const mockMetaMixin = <T extends Constructor<WidgetBase<any>>>(Base: T, mockStub: jest.Mock): T => {
```
Comment (bitpshr) [inline src/pages/Examples.spec.tsx]: We should think about optimizing `Block`-based testing at some point. For now, `Block`-powered widgets each mock their own meta, which works, but we could probably unify this into a nice helper API.

File: src/pages/Examples.spec.tsx [hunk 3]
```diff
@@ -1,16 +1,94 @@
...
+			{
+				code: 'code',
+				demo: 'demo',
+				example: 'example',
+				exampleName: 'name',
+				overview: 'overview',
+				sandbox: true
+			}
+		]);
+		const h = harness(() => w(mockMetaMixin(Examples, mockMeta), {}));
```
Comment (agubler) [inline src/pages/Examples.spec.tsx]: Should be able to use `tsx` if you create do something like 
```ts
const ExampleWithMock = mockMetaMixin(Examples, mockMeta);
``` 
before using it 
```ts
const h = harness(() => <ExampleWithMock />);
```

(you don't have to use my rubbish name though!)

File: src/widgets/Landing.tsx [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import WidgetBase from '@dojo/framework/widget-core/WidgetBase';
+import { tsx } from '@dojo/framework/widget-core/tsx';
+import { WidgetProperties } from '@dojo/framework/widget-core/interfaces';
+import * as css from './Landing.m.css';
+
+export default class Landing extends WidgetBase<WidgetProperties> {
```
Comment (bitpshr) [inline src/widgets/Landing.tsx]: A standalone `Landing` widget was created so this styling (and potential functionality in the future) could be used outside of the content pipeline.

File: src/widgets/section/Subsection.tsx [hunk 5]
```diff
@@ -0,0 +1,11 @@
+import WidgetBase from '@dojo/framework/widget-core/WidgetBase';
+import { WidgetProperties } from '@dojo/framework/widget-core/interfaces';
+import { tsx } from '@dojo/framework/widget-core/tsx';
+
+import * as css from './Subsection.m.css';
+
+export default class Subsection extends WidgetBase<WidgetProperties> {
```
Comment (bitpshr) [inline src/widgets/section/Subsection.tsx]: A standalone `Subsection` widget was created so this styling (and potential functionality in the future) could be used outside of the content pipeline.

File: src/pages/Examples.tsx [hunk 6]
```diff
@@ -1,14 +1,65 @@
...
+				</a>
+			)}
+			<a href={`https://github.com/dojo/examples/tree/master/${exampleName}`} classes={css.linkBtn}>
+				<FontAwesomeIcon icon="code-branch" />
+			</a>
+		</div>
+	);
+
 	protected render() {
+		const examples = (this.meta(Block).run(getExamples)() as any) as ExampleMeta[];
```
Comment (agubler) [inline src/pages/Examples.tsx]: I think this can be `null` too as meta's will return nothing until the block as resolved with the value.

Comment (agubler) [reply src/pages/Examples.tsx]: I'm surprised this doesn't break though, if it does return null with the `.map` below.

Comment (bitpshr) [reply src/pages/Examples.tsx]: That probably explains why I thought the dev server was broken sometimes.

**Pull request merged**